PLAT-40278 : Add background circle in Checkbox

changgilee opened the pull request:
Add background circle base on `18 GUI guidance in Enyo.

Enyo-DCO-1.1-Signed-off-by: Changgi Lee <changgi.lee@lge.com>

changgilee committed:
PLAT-40278 : Add background circle in Checkbox

Add background circle base on `18 GUI guidance in Enyo.

Enyo-DCO-1.1-Signed-off-by: Changgi Lee <changgi.lee@lge.com>

changgilee committed:
PLAT-40278: Add neutral style

Add nutural style in Checkbox

Enyo-DCO-1.1-Signed-off-by: Changgi Lee <changgi.lee@lge.com>

yeram commented on src/Checkbox/Checkbox.less:
```diff
@@ -2,22 +2,74 @@
 		padding: 0;
+		width: 30px;
+		height: 30px;
+		border-radius: 15px;
```
Based on `enact` we also need `line-height` as `30px` and `font-size` as `60px`.

yeram committed:
Add missing styles and refactor some

Add line-height and font-size for check icon alignment.
Add refactor some less definition for better understanding.

yeram committed:
Reverse toggle item style based on checked state

It was wrong definition.

Djspaceg commented on src/Checkbox/Checkbox.less:
```diff
@@ -2,22 +2,73 @@
 		margin: 0;
 		padding: 0;
+		width: 30px;
+		height: 30px;
```
These would probably be better as a `@moon-checkbox-size` variable in the `variables.less` file

Djspaceg commented on src/Checkbox/Checkbox.less:
```diff
@@ -2,22 +2,73 @@
+		width: 30px;
+		height: 30px;
+		line-height: 30px;
+		font-size: 60px;
```
This line could become `font-size: (@moon-checkbox-size * 2);`

Djspaceg commented on src/Checkbox/Checkbox.less:
```diff
@@ -2,22 +2,73 @@
+		height: 30px;
+		line-height: 30px;
+		font-size: 60px;
+		border-radius: 15px;
```
This line could become `font-size: (@moon-checkbox-size / 2);`

Djspaceg commented on src/Checkbox/Checkbox.less:
```diff
@@ -2,22 +2,73 @@
+		font-size: 60px;
+		border-radius: 15px;
+		color: transparent;
+		background-color: @moon-dark-gray;
```
It's best practice to not refer to these variables outside of the `moonstone-dark/light.less` files, and instead assign the `@moon-dark-gray` to a more descriptive color variable name, like `@moon-checkbox-bg-color: @moon-dark-gray;`. This is simpler and easier to maintain over time and abstracts the color value from the component LESS files. Be sure to update both the light and dark color files.

Djspaceg commented on src/Checkbox/Checkbox.less:
```diff
@@ -2,22 +2,73 @@
+
+	.moon-checkbox:not([disabled]) {
+		.moon-icon {
+			background-color: rgba(255,255,255,0.41);
```
This should be a color variable too. Also, do you think anyone can notice the difference between `0.41` and `0.4` alpha levels? Let's go ahead and round this value down to `0.4`: `rgba(255,255,255,0.4)`.
Alternatively, in the color file, you could use the LESS function to apply the alpha as well: `fade(@moon-white, 40%)`

Djspaceg reviewed (changes requested):
Let's make a few organizational changes to this PR.

yeram committed:
Use less variable and function

Pull out common props to less variable and use them.
Use less function for rgba color.

yeram committed:
Modify font-size, line-height for formcheckbox and toggleitem

It should not be changed when the parent is formcheckbox or toggleitem

Djspaceg commented on src/Checkbox/Checkbox.less:
```diff
@@ -2,22 +2,73 @@
+			color: transparent;
+		}
+		&[checked] .moon-icon {
+			color: @moon-white;
```
Each of these `@moon-white` variables should be updated to be `@moon-spotlight-text-color`. This one, line 67, and 46.

yeram committed:
Use color abstraction variable

Change @moon-white to @moon-spotlight-text-color.
